tests/integration/azure.py

```diff
@@ -583,12 +583,17 @@ def upload_image(self, progress_function = None):
 
             image_uri = f"https://{self._storageaccount.name}.blob.core.windows.net/vhds/{image_name}.vhd"
 
+            if "hyper_v_generation" in self.config:
+                hyper_v_generation = self.config["hyper_v_generation"]
```

Comment:
What happens with an unexpected value like `V3`? Maybe we should validate for allowed values?

Reply:
I agree with this. We should probably do something like this:
```
allowed_generations = ["V1", "V2"]
hyper_v_generation = self.config.get("hyper_v_generation", allowed_generations[0])
if hyper_v_generation not in allowed_generations:
   raise RuntimeError(f"Hypervisor generation '{hyper_v_generation}' not supported ({allowed_generations})") 
```